client/hotrod-client/src/main/java/org/infinispan/client/hotrod/counter/operation/BaseCounterOperation.java

```diff
@@ -0,0 +1,73 @@
+package org.infinispan.client.hotrod.counter.operation;
+
+import java.net.SocketAddress;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.Configuration;
+import org.infinispan.client.hotrod.impl.operations.RetryOnFailureOperation;
+import org.infinispan.client.hotrod.impl.protocol.Codec;
+import org.infinispan.client.hotrod.impl.protocol.HeaderParams;
+import org.infinispan.client.hotrod.impl.transport.Transport;
+import org.infinispan.client.hotrod.impl.transport.TransportFactory;
+import org.infinispan.client.hotrod.logging.LogFactory;
+import org.infinispan.commons.logging.Log;
+import org.infinispan.counter.exception.CounterException;
+
+/**
+ * A base operation class for the counter's operation.
+ *
+ * @author Pedro Ruivo
+ * @since 9.2
+ */
+abstract class BaseCounterOperation<T> extends RetryOnFailureOperation<T> {
+
+   private static final Log commonsLog = LogFactory.getLog(BaseCounterOperation.class, Log.class);
+   private static final byte[] EMPTY_CACHE_NAME = new byte[0];
+   private static final byte[] COUNTER_CACHE_NAME = RemoteCacheManager.cacheNameBytes("org.infinispan.counter");
+   private final String counterName;
+
+   BaseCounterOperation(Codec codec, TransportFactory transportFactory, AtomicInteger topologyId, Configuration cfg,
+         String counterName) {
+      super(codec, transportFactory, EMPTY_CACHE_NAME, topologyId, 0, cfg);
```

Comment:
The counters operation can be invoked in any node in the cluster. However, the TransportFactory needs a cache name. I didn't want to add complexity and implement a counter transport factory so I'm setting the cache name to update the list of servers in the factory.

Reply:
1st, `TransportFactory.getTransport(Set<SocketAddress> failedServers, byte[] cacheName);` < needs the cache name!
2nd, I don't want to expose any internals. If we change the counters to use anything else than a cache, I don't want to break the clients! 
3rd, if an use implements the protocol in its own language, I don't want them to know that they suppose to use a specific cache name.

Reply:
My bad, I haven't noticed that you don't use the cache name in the request header, only when reading the response. Besides not exposing the internals, it makes sense to save those bytes in the request.
It would make sense to drop the cache name from the protocol completely, though the complexity of parsing (based on opCode as flags come after cache name) is probably not worth the saved single byte.

Reply:
I prefer to have the cache name there too.